### PR TITLE
TinyMCE 6+ support

### LINKED ISSUE #106: Support for TinyMCE 6?
Hi there,

are there any plans to support TinyMCE 6? In TnyMCE 6 the `window.tinymce.util.XHR` object was removed [1]. We access it in protect.js line 32 which fails with an 

```
Uncaught TypeError: Cannot read properties of undefined (reading '_send')
    at ++resource++protect.js:32:48
```

[1] https://www.tiny.cloud/docs/api/tinymce.util/tinymce.util.xhr/

### PR DESCRIPTION
**Pull request by petschki**
check for removed `tinymce.util.XHR`

EDIT: Need to check how to patch the new TinyMCE fetch implementation.

fixes #106

### CODE REVIEW

**Comment by petschki** [comment]
@jenkins-plone-org please run jobs

**Review by davisagli — COMMENTED** [review]
This code was patching TinyMCE to add the Plone CSRF authenticator when tinymce makes requests. Is that no longer necessary?

**Comment by petschki** [comment]
hm, you're right. TinyMCE refactored all XHR calls to fetch API ... need to check how we can patch this.

File: plone/protect/protect.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ if(script){
       }
     });
   }
-  if(window.tinymce && window.tinymce.util.XHR._send === undefined){
+  if(window.tinymce && window.tinymce.util.XHR && window.tinymce.util.XHR._send === undefined){
```
Comment (thet) [inline plone/protect/protect.js:32]:
```suggestion
  if(window.tinymce && window.tinymce.util.XHR && window.tinymce.util.XHR._send === undefined){
    // Fix for TinyMCE < 5 in Plone < 6.
    // TODO: Can be removed in Plone 6.1
```

**Review by thet — APPROVED** [review]
LGTM
This is a very unobtrusive change which doesn't break anything but fix the current situation in Plone 6.

I‌ added a suggestion comment to indicate that this dead code can be removed with Plone 6.1 or whatever.

**Comment by petschki** [comment]
Just to be clear: this fix is for TinyMCE 6+ ... Plone 6.0 officially ships with TinyMCE 6.0 and therefore needs this fix. I've changed the commend accordingly.

**Comment by petschki** [comment]
@jenkins-plone-org please run jobs

**Review by mauritsvanrees — APPROVED** [review]

**Comment by mauritsvanrees** [comment]
Jenkins failures are unrelated.  Can be merged, but Jenkins is really busy now.

**Pull request merged**